src/compiler/transformers/esDecorators.ts

```diff
@@ -990,37 +1015,64 @@ export function transformESDecorators(context: TransformationContext): (x: Sourc
 
     function visitClassDeclaration(node: ClassDeclaration): VisitResult<Statement> {
         if (isDecoratedClassLike(node)) {
-            if (hasSyntacticModifier(node, ModifierFlags.Export) &&
-                hasSyntacticModifier(node, ModifierFlags.Default)) {
-                //  export default (() => { ... })();
-                const originalClass = getOriginalNode(node, isClassLike) ?? node;
-                const className = originalClass.name ? factory.createStringLiteralFromNode(originalClass.name) : factory.createStringLiteral("default");
-                const iife = transformClassLike(node, className);
-                const statement = factory.createExportDefault(iife);
-                setOriginalNode(statement, node);
-                setCommentRange(statement, getCommentRange(node));
-                setSourceMapRange(statement, moveRangePastDecorators(node));
-                return statement;
+            const statements: Statement[] = [];
+            const originalClass = getOriginalNode(node, isClassLike) ?? node;
+            const className = originalClass.name ? factory.createStringLiteralFromNode(originalClass.name) : factory.createStringLiteral("default");
+            const isExport = hasSyntacticModifier(node, ModifierFlags.Export);
+            const isDefault = hasSyntacticModifier(node, ModifierFlags.Default);
+            if (!node.name) {
+                node = injectClassNamedEvaluationHelperBlockIfMissing(context, node, className);
+            }
+            if (isExport && isDefault) {
+                const iife = transformClassLike(node);
+                if (node.name) {
+                    //  let C = (() => { ... })();
+                    //  export default C;
+                    const varDecl = factory.createVariableDeclaration(factory.getLocalName(node), /*exclamationToken*/ undefined, /*type*/ undefined, iife);

```

Comment:
does transformClassLike not apply now because the order of esDecorators has changed?

Reply:
Yeah, I think I misread this earlier.